lua/entities/gmod_wire_egp/lib/egplib/transmitreceive.lua

```diff
@@ -619,19 +619,31 @@ if (SERVER) then
 					DataToSend[#DataToSend+1] = { ID = obj.ID, index = obj.index, Settings = obj:DataStreamInfo() }
 				end
 
-				timer.Simple( k, function() -- send 1 second apart
+				timer.Simple( k - 1, function() -- send 1 second apart, send the first one instantly
 					local isLastScreen = ((k == #targets) and #targets or nil)
 					if silent then
 						isLastScreen = nil
 					end
 
+					local data = {
+						Ent = v,
+						Objects = DataToSend,
+						Filtering = v.GPU_texture_filtering,
+						IsLastScreen = isLastScreen -- Doubles as notifying the client that no more data will arrive, and tells them how many did arrive
+					}
+
+					local von = WireLib.von.serialize(data)
+					local compressed = util.Compress(von)
+					local compressedLength = #compressed
+
+					if compressedLength > 60000 then
```

Comment:
Should check the uncompressed length, and this length should also be used in the decompress.